src/txn/txn.c

```diff
@@ -804,7 +811,7 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
 	update_timestamp = F_ISSET(txn, WT_TXN_HAS_TS_COMMIT);
 #endif
 
-	__wt_txn_release(session);
+	WT_ERR(__wt_txn_release(session));
```

Comment:
I don't think this really is valid here.  We are well beyond the point in {{wt_txn_commit}} where it states "Note: we're going to commit: nothing can fail after this point.".  What is a user going to do with an error return from {{wt_txn_release}}?  Should they assume their transaction/data is not committed? It is committed and visible yet the commit returned error.